packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -190,11 +201,226 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const searchInputRef = useRef<HTMLInputElement>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const [isMineOnly, setIsMineOnly] = useState(false);
+    const [searchText, setSearchText] = useState('');
+    const closeQueryImporter = (): void => {
+      editorStore.queryLoaderState.setIsQueryLoaderOpen(false);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const toggleIsMineOnly = (): void => {
+      setIsMineOnly(!isMineOnly);
+    };
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.queryLoaderState.setIsQueryLoaderOpen(false);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
```

Comment:
let's use `jumpTo` so you don't need to do `window.location.reload()`. We use methods from `navigator` and rarely poke directly into the `window` object. Also, I think we don't need `setBlockingAlert` here

Reply:
re: setBlockingAlert.  I added it for this previous comment. 
<img width="719" alt="Screen Shot 2022-09-13 at 11 49 32 AM" src="https://user-images.githubusercontent.com/73408381/189947802-8f0d9e6d-f95d-4567-85c6-912a493f2624.png">
May I delete it?